src/fu-remotes.c

```diff
@@ -0,0 +1,504 @@
+/*
+ * Copyright (C) 2017-2019 Richard Hughes <richard@hughsie.com>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#define G_LOG_DOMAIN				"FuRemotes"
+
+#include "config.h"
+
+#include <xmlb.h>
+#include <glib-object.h>
+#include <gio/gio.h>
+#include <glib/gi18n.h>
+
+#include "fu-common.h"
+#include "fu-remotes.h"
+
+#include "fwupd-common.h"
+#include "fwupd-error.h"
+#include "fwupd-remote-private.h"
+
+enum {
+	SIGNAL_CHANGED,
+	SIGNAL_LAST
+};
+
+static guint signals[SIGNAL_LAST] = { 0 };
+
+static void fu_remotes_finalize	 (GObject *obj);
+
+struct _FuRemotes
+{
+	GObject			 parent_instance;
+	GPtrArray		*remotes;		/* (element-type FwupdRemote) */
+	GPtrArray		*monitors;		/* (element-type GFileMonitor) */
+	XbSilo			*silo;
+	GHashTable		*os_release;
+};
+
+G_DEFINE_TYPE (FuRemotes, fu_remotes, G_TYPE_OBJECT)
+
+static void
+fu_remotes_emit_changed (FuRemotes *self)
+{
+	g_debug ("::remotesuration changed");
```

Comment:
what exactly is a `remotesuration`?

Reply:
I'm guessing a find and replace of `config` to `remote` that applied to `configuration`?

Reply:
It's an old 16 century English word.... I'm bullshitting. It's search and replace :) Fixed.